platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/components/transaction/system/internal/PostConsensusSystemTransactionManagerImpl.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (C) 2016-2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.components.transaction.system.internal;
+
+import static com.swirlds.logging.LogMarker.EXCEPTION;
+
+import com.swirlds.common.system.transaction.internal.SystemTransaction;
+import com.swirlds.platform.components.transaction.system.PostConsensusSystemTransactionHandler;
+import com.swirlds.platform.components.transaction.system.PostConsensusSystemTransactionManager;
+import com.swirlds.platform.components.transaction.system.PostConsensusSystemTransactionTypedHandler;
+import com.swirlds.platform.internal.ConsensusRound;
+import com.swirlds.platform.internal.EventImpl;
+import com.swirlds.platform.state.State;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * Implementation of {@link PostConsensusSystemTransactionManager}
+ */
+public class PostConsensusSystemTransactionManagerImpl implements PostConsensusSystemTransactionManager {
```

Comment:
Not a requirement for this PR. Eventually we may want to consider instrumenting this class with some metrics.